src/sentry/models/authprovider.py

```diff
@@ -48,5 +61,87 @@ def get_provider(self):
 
         return manager.get(self.provider, **self.config)
 
+    def get_scim_token(self):
+        from sentry.models import SentryAppInstallationForProvider
+
+        if self.flags.scim_enabled:
+            return SentryAppInstallationForProvider.get_token(
+                self.organization, f"{self.provider}_scim"
+            )
+        else:
+            logger.warning(
+                "SCIM disabled but tried to access token",
+                extra={"organization_id": self.organization.id},
+            )
+            return None
+
+    def get_scim_url(self):
+        if self.flags.scim_enabled:
+            return f"https://sentry.io/scim/{self.organization.slug}/scim/v2"
+            # TODO: make dynamic once URL routes are added
+        else:
+            return None
+
+    def enable_scim(self, user):
+        from sentry.mediators.sentry_apps import InternalCreator
+        from sentry.models import SentryAppInstallation, SentryAppInstallationForProvider
+
+        if (
+            not self.get_provider().can_use_scim(self.organization, user)
+            or self.flags.scim_enabled is True
+        ):
+            logger.warning(
+                "SCIM already enabled",
+                extra={"organization_id": self.organization.id},
+            )
+            return
+
+        # check if we have a scim app already
+
+        if SentryAppInstallationForProvider.objects.filter(
+            organization=self.organization, provider="okta_scim"
+        ).exists():
+            logger.warning(
+                "SCIM instillation already exists",
+                extra={"organization_id": self.organization.id},
+            )
+            return
+
+        data = {
+            "name": "SCIM Internal Integration",
+            "author": "Auto-generated by Sentry",
+            "organization": self.organization,
+            "overview": SCIM_INTERNAL_INTEGRATION_OVERVIEW,
+            "user": user,
+            "scopes": [
```

Comment:
wonder if we could lock this down a bit more -- this token doesn't need all of what org:admin gives, for example, but needed to edit teams. seems like we could either 1) create a scim-speciifc scope, or create more granular scopes within scope categories

Reply:
👍🏼 will think about this some more and put some potential paths forward in the spec doc.

Reply:
I actually misunderstood a bit how our permissions scopes work -- an actor's scope only needs one of the scopes in the array in order to satisfy the permission, not all of them. I think we can remove the majority of these and have it only be `member:admin`, `team:write` and `org:read`. will do so in a follow up PR.